File: v05/zad3/Trougao.cpp
#include "Trougao.hpp"

// Konstruktor sa parametrima
Trougao::Trougao(double a, double b, double c) {
    this->a=a;
    this->b=b;
    this->c=c;
}

// Konstruktor kopije
Trougao::Trougao(const Trougao &t) {
    a=t.a;
    b=t.b;
    c=t.c;
}

// Geteri
double Trougao::getA() const {
    return a;
}

double Trougao::getB() const {
    return b;
}

double Trougao::getC() const {
    return c;
}

// Seteri
void Trougao::setA(double a) {
    this->a=a;
}

void Trougao::setB(double b) {
    this->b=b;
}

void Trougao::setC(double c) {
    this->c=c;
}

// Vraća obim trougla
double Trougao::getO() const {
    return a+b+c;
}

// Vraća površinu trougla
double Trougao::getP() const {
    double s=(a+b+c)/2;
    return sqrt(s*(s-a)*(s-b)*(s-c));
}

File: v10/zad5_1/Trougao.java
package zad5_1;

public class Trougao implements Figura {
	
	// Polja
	private double a;
	private double b;
	private double c;
	
	// Konstruktor sa parametrima (izgenerisan)
	public Trougao(double a, double b, double c) {
		super();
		this.a = a;
		this.b = b;
		this.c = c;
	}
	
	// Geteri i seteri (izgenerisani)
	public double getA() {
		return a;
	}

	public void setA(double a) {
		this.a = a;
	}

	public double getB() {
		return b;
	}

	public void setB(double b) {
		this.b = b;
	}
	
	public double getC() {
		return c;
	}

	public void setC(double c) {
		this.c = c;
	}
	
	@Override
	public double obim() {
		return a + b + c;
	}
	
	@Override
	public double povrsina() {
		double s = obim() / 2;
		return Math.sqrt(s * (s - a) * (s - b) * (s - c));
	}

	@Override
	public String toString() {
		return "Trougao [a=" + a + ", b=" + b + ", c=" + c + "]";
	}
	
}
